### PR TITLE
Ava initial commit

### PR DESCRIPTION
**Pull request by SneakboxKB**
Initial AVA firmware commit

First firmware submission for AVA keyboard

- [X] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [X] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sneakbox/ava/ava.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(D6, led_state.caps_lock);
+        writePin(D4, led_state.num_lock);
+    }
+        writePin(D7, 1);
+    return res;
+}
```
Comment (drashna) [inline keyboards/sneakbox/ava/ava.c:36]:
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index,clockwise)) { return false; }
    if (index == 0) { /* First encoder */
        if (clockwise) {
            tap_code(KC_DOWN);
        } else {
            tap_code(KC_UP);
        }
    } else if (index == 1) { /* Second encoder */
        if (clockwise) {
            tap_code(KC_DOWN);
        } else {
            tap_code(KC_UP);
        }
    }
    return true;
}

File: keyboards/sneakbox/ava/config.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define RGB_DI_PIN F7
```
Comment (drashna) [inline keyboards/sneakbox/ava/config.h:60]:
```suggestion
#define LED_NUM_LOCK_PIN  D4
#define LED_CAPS_LOCK_PIN D6

#define RGB_DI_PIN F7
```

File: keyboards/sneakbox/ava/config.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
Comment (drashna) [inline keyboards/sneakbox/ava/config.h:58]:
```suggestion
```

File: keyboards/sneakbox/ava/config.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+#define RGB_DI_PIN F7
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/sneakbox/ava/config.h:61]: could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/sneakbox/ava/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            tap_code(KC_UP);
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_DOWN);
+        } else {
+            tap_code(KC_UP);
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/sneakbox/ava/keymaps/default/keymap.c:55]:
```suggestion
```

File: keyboards/sneakbox/ava/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            tap_code(KC_VOLU);
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/sneakbox/ava/keymaps/via/keymap.c:70]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by SneakboxKB — COMMENTED** [review]
Reviewed and agree. Thanks!

File: keyboards/sneakbox/ava/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# Sneakbox AVA
+
+![Sneakbox AVA](https://i.imgur.com/lVaAmfS.jpg)
```
Comment (fauxpark) [inline keyboards/sneakbox/ava/readme.md:3]:
```suggestion
![Sneakbox AVA](https://i.imgur.com/lVaAmfSh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/sneakbox/ava/info.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
+﻿{
+  "keyboard_name": "AVA",
+  "url": "https://sneakbox.com",
+  "maintainer": "mujimanic",
+  "width": 18.75,
+  "height": 5,
+  "layouts": {
+      "LAYOUT_ava_split_bs": {
```
Comment (fauxpark) [inline keyboards/sneakbox/ava/info.json:8]:
```suggestion
      "LAYOUT_alice_split_bs": {
```

File: keyboards/sneakbox/ava/info.json [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+              {"x":0, "y":1}, {"x":1.25, "y":1, "w":1.5}, {"x":2.75, "y":1}, {"x":3.75, "y":1}, {"x":4.75, "y":1}, {"x":5.75, "y":1}, {"x":6.75, "y":1}, {"x":9.75, "y":1}, {"x":10.75, "y":1}, {"x":11.75, "y":1}, {"x":12.75, "y":1}, {"x":13.75, "y":1}, {"x":14.75, "y":1}, {"x":15.75, "y":1}, {"x":16.75, "y":1, "w":1.5},
+              {"x":0, "y":2}, {"x":1.25, "y":2, "w":1.75}, {"x":3, "y":2}, {"x":4, "y":2}, {"x":5, "y":2}, {"x":6, "y":2}, {"x":7, "y":2}, {"x":10, "y":2}, {"x":11, "y":2}, {"x":12, "y":2}, {"x":13, "y":2}, {"x":14, "y":2}, {"x":15, "y":2}, {"x":16, "y":2, "w":2.25},
+              {"x":0, "y":3}, {"x":1.25, "y":3, "w":2.25}, {"x":3.5, "y":3}, {"x":4.5, "y":3}, {"x":5.5, "y":3}, {"x":6.5, "y":3}, {"x":7.5, "y":3}, {"x":9.5, "y":3}, {"x":10.5, "y":3}, {"x":11.5, "y":3}, {"x":12.5, "y":3}, {"x":13.5, "y":3}, {"x":14.5, "y":3}, {"x":15.65, "y":3.25}, {"x":16.75, "y":3, "w":1.5},
+              {"x":1.25, "y":4, "w":1.5}, {"x":2.75, "y":4}, {"x":5, "y":4, "w":1.5}, {"x":6.5, "y":4, "w":2.25}, {"x":9.75, "y":4, "w":2.75}, {"x":12.5, "y":4, "w":1}, {"x":14.65, "y":4.25}, {"x":15.65, "y":4.25}, {"x":16.65, "y":4.25}
+			]
+      },
+      "LAYOUT_ava": {
```
Comment (fauxpark) [inline keyboards/sneakbox/ava/info.json:17]:
```suggestion
      "LAYOUT_alice": {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**